cidsets/cidsets_test.go

```diff
@@ -18,30 +18,142 @@ func TestCIDSetManager(t *testing.T) {
 	setID1 := SetID("set1")
 	setID2 := SetID("set2")
 
+	// set1: +cid1
 	exists, err := mgr.InsertSetCID(setID1, cid1)
 	require.NoError(t, err)
 	require.False(t, exists)
 
+	// set1: +cid1 (again)
 	exists, err = mgr.InsertSetCID(setID1, cid1)
 	require.NoError(t, err)
 	require.True(t, exists)
 
+	// set2: +cid1
 	exists, err = mgr.InsertSetCID(setID2, cid1)
 	require.NoError(t, err)
 	require.False(t, exists)
 
+	// set2: +cid2 (again)
 	exists, err = mgr.InsertSetCID(setID2, cid1)
 	require.NoError(t, err)
 	require.True(t, exists)
 
+	// delete set1
 	err = mgr.DeleteSet(setID1)
 	require.NoError(t, err)
 
+	// set1: +cid1
 	exists, err = mgr.InsertSetCID(setID1, cid1)
 	require.NoError(t, err)
 	require.False(t, exists)
 
+	// set1: +cid1 (again)
 	exists, err = mgr.InsertSetCID(setID2, cid1)
 	require.NoError(t, err)
 	require.True(t, exists)
 }
+
+func TestCIDSetToArray(t *testing.T) {
+	cids := testutil.GenerateCids(2)
+	cid1 := cids[0]
+	cid2 := cids[1]
+
+	dstore := ds_sync.MutexWrap(ds.NewMapDatastore())
+	mgr := NewCIDSetManager(dstore)
+	setID1 := SetID("set1")
+
+	// Expect no items in set
+	len, err := mgr.SetLen(setID1)
+	require.NoError(t, err)
+	require.Equal(t, 0, len)
+
+	arr, err := mgr.SetToArray(setID1)
+	require.NoError(t, err)
+	require.Len(t, arr, 0)
+
+	// set1: +cid1
+	exists, err := mgr.InsertSetCID(setID1, cid1)
+	require.NoError(t, err)
+	require.False(t, exists)
+
+	// Expect 1 cid in set
+	len, err = mgr.SetLen(setID1)
+	require.NoError(t, err)
+	require.Equal(t, 1, len)
+
+	arr, err = mgr.SetToArray(setID1)
+	require.NoError(t, err)
+	require.Len(t, arr, 1)
+	require.Equal(t, arr[0], cid1)
+
+	// set1: +cid1 (again)
+	exists, err = mgr.InsertSetCID(setID1, cid1)
+	require.NoError(t, err)
+	require.True(t, exists)
+
+	// Expect 1 cid in set
+	len, err = mgr.SetLen(setID1)
+	require.NoError(t, err)
+	require.Equal(t, 1, len)
+
+	arr, err = mgr.SetToArray(setID1)
+	require.NoError(t, err)
+	require.Len(t, arr, 1)
+	require.Equal(t, arr[0], cid1)
+
+	// set1: +cid2
+	exists, err = mgr.InsertSetCID(setID1, cid2)
+	require.NoError(t, err)
+	require.False(t, exists)
+
+	// Expect 2 cids in set
+	len, err = mgr.SetLen(setID1)
+	require.NoError(t, err)
+	require.Equal(t, 2, len)
+
+	arr, err = mgr.SetToArray(setID1)
+	require.NoError(t, err)
+	require.Len(t, arr, 2)
+	require.Contains(t, arr, cid1)
+	require.Contains(t, arr, cid2)
+
+	// Delete set1
+	err = mgr.DeleteSet(setID1)
+	require.NoError(t, err)
+
+	// Expect no items in set
+	len, err = mgr.SetLen(setID1)
+	require.NoError(t, err)
+	require.Equal(t, 0, len)
+
+	arr, err = mgr.SetToArray(setID1)
+	require.NoError(t, err)
+	require.Len(t, arr, 0)
```

Comment:
@dirkmc One more important thing to test here:

-> Create the manager again using `NewCIDSetManager ` with the same data-store.
-> Read length -> should work.
-> Insert and read again -> should work.

This is to ensure things work correctly on a restart.